distributed/dashboard/components/scheduler.py

```diff
@@ -559,6 +559,140 @@ def update(self):
+            )
+            self.memory_figure.add_tools(hover)
+            self.memory_figure.yaxis[0].formatter = NumeralTickFormatter(format="0.0 b")
+            self.memory_figure.xgrid.visible = False
+
+            # Network
+            self.network_source = ColumnDataSource(
+                {
+                    "size": [],
+                    "bandwidth": [],
+                }
+            )
+
+            self.network_figure = figure(
+                title="Network Bandwidth -- Computing ...",
+                tools="",
+                toolbar_location="above",
+                x_range=["1 kiB", "10kiB", "100kiB", "1 MiB", "10 MiB", "50 MiB"],
+                **kwargs,
+            )
+
+            self.network_figure.vbar(
+                x="size", top="bandwidth", width=0.9, source=self.network_source
+            )
+            hover = HoverTool(
+                mode="vline", tooltips=[("Bandwidth", "@bandwidth{0.00 b}/s")]
+            )
+            self.network_figure.add_tools(hover)
+            self.network_figure.yaxis[0].formatter = NumeralTickFormatter(
+                format="0.0 b"
+            )
+            self.network_figure.xgrid.visible = False
+
+            self.root = row(
+                self.memory_figure,
+                self.disk_figure,
+                self.network_figure,
+            )
+
+            self.memory_data = {
+                "size": [],
+                "bandwidth": [],
+            }
+            self.disk_data = {
+                "size": [],
+                "bandwidth": [],
+            }
+            self.network_data = {
+                "size": [],
+                "bandwidth": [],
+            }
+
+            async def f():
+                result = await self.scheduler.benchmark_hardware()
+
+                for size, bandwidth in result["disk"].items():
+                    self.disk_data["size"].append(size)
+                    self.disk_data["bandwidth"].append(bandwidth)
+
+                for size, bandwidth in result["memory"].items():
+                    self.memory_data["size"].append(size)
+                    self.memory_data["bandwidth"].append(bandwidth)
+
+                for size, bandwidth in result["network"].items():
+                    self.network_data["size"].append(size)
+                    self.network_data["bandwidth"].append(bandwidth)
+
+            self.scheduler.loop.add_callback(f)
+
+    def update(self):
+        if any(self.disk_data.values()):
+            if self.memory_figure.title.text == "Memory Bandwidth":
+                return
+            else:
+                update(self.disk_source, self.disk_data)
+                update(self.memory_source, self.memory_data)
+                update(self.network_source, self.network_data)
+                self.memory_figure.title.text = "Memory Bandwidth"
+                self.disk_figure.title.text = "Disk Bandwidth"
+                self.network_figure.title.text = "Network Bandwidth"
```

Comment:
```suggestion
        if not self.disk_data["size"]:
            return

        update(self.disk_source, self.disk_data)
        update(self.memory_source, self.memory_data)
        update(self.network_source, self.network_data)
        self.disk_figure.title.text = "Disk Bandwidth"
        self.memory_figure.title.text = "Memory Bandwidth"
        self.network_figure.title.text = "Network Bandwidth"
```

Reply:
Can I ask for the motivation for this change?

Reply:
code readability

Reply:
This would also cause a change where we update at every event, which probably isn't desired or useful.  We can do it though.  I don't have a strong preference here (it's cheap to do).  This isn't just a style change though.  Happy either way.

Reply:
It should not have any functional impact. ``self.disk_data["size"]`` will be empty until and only until your async task is finished. Also testing for the title is redundant.

Reply:
After the async task is finished then the following code will be run every update.  Previously it was run only once.  

Anyway, I'm just going to go ahead with this.  It's not that important either way I don't think.

Reply:
Ah. I fixed my suggestion.